pkg/shared/util/grpc_utils.go

```diff
@@ -20,18 +20,89 @@ import (
 	"context"
 	"fmt"
 	"log"
+	"os"
+	"strings"
 	"time"
 
+	"github.com/Masterminds/semver/v3"
+	pep440 "github.com/aquasecurity/go-pep440-version"
 	"github.com/numaproj/numaflow-go/pkg/info"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/status"
 
+	"github.com/numaproj/numaflow"
 	sdkerr "github.com/numaproj/numaflow/pkg/sdkclient/error"
 	resolver "github.com/numaproj/numaflow/pkg/sdkclient/grpc_resolver"
 )
 
+func checkConstraint(version *semver.Version, constraint string) error {
+	if c, err := semver.NewConstraint(constraint); err != nil {
+		return fmt.Errorf("error parsing constraint: %w, constraint string: %s", err, constraint)
+	} else if isValid, _ := c.Validate(version); !isValid {
+		return fmt.Errorf("version %v did not meet constraint requirement %s", version, constraint)
+	}
+	return nil
+}
+
+// checkNumaflowCompatibility checks if the current numaflow version is compatible with the given language's SDK version
+func checkNumaflowCompatibility(numaflowVersion string, minNumaflowVersion string) error {
+	// Check if server info contains MinimumNumaflowVersion
+	if minNumaflowVersion == "" {
+		return fmt.Errorf("server info does not contain minimum numaflow version. Upgrade SDK to latest version")
+	}
+
```

Comment:
We can remove this piece of code. It should never be "", and even if it is, it will be caught by line 62.

Reply:
if we get rid of this then in the error on line 62 message the minNumaflowVersion will be blank and might confuse users

Reply:
It won't, right? user should immediately realize that the minNumaflowVersion is empty

Reply:
You are right, we should check in the case when user use an old sdk version which don't support writing minNumaflowVersion. Please also surface to the user which exact version to upgrade to instead of latest. Because latest might not be compatible to current numaflow.